content/articles/decision-making-and branching-in-java/index.md

```diff
@@ -0,0 +1,279 @@
+    }
+}
+```
+
+When a number is entered a test is carried if the modulo is zero the output is given as even or else the output is odd.
+
+### Nested if...else statement
+When a series of decisions are to be involved we use the nested if-else statement and the if-else statements are used more than once.
+
+```Java
+if(test condition a)
+{
+    if(test condition b)
+    {
+        statement-1;
+    }
+    else
+    {
+    statement-2;
+    }
+}
+else
+{
+    statement-3;
+}
+statement-x;
+```
+
+If the test condition a is false then statement-3 is executed otherwise if test condition a is true it proceeds to test condition b which if it is true the statement-1 is executed and if false the statement-2 is executed.
+When nesting, the if...else statements every if should be matched with an else statement. In nesting if...else statements in java you consider an else as linked to the closest non-terminated if.
+Using the nested if...else in an example:
+
+```Java
+public class Largest {
+  public static void main(String[] args) {
+        int i=40, j=23, k=50;
+        System.out.println("largest value is:");
+        if (i>j)
+        {
+            if (i>k)
+            {
+                System.out.println(i);
+            }
+            else
+            {
+         System.out.println(k);
+            }
+        }
+        else
+        {
+            if(j>k)
+            {
+                System.out.println(k);
+            }
+            else
+            {
+                
+                System.out.println(j);
+            }
+        }
+    }
+}
+```
+The system will display the largest number of the three that is 50.
+
+### The else..if ladder
+The else-if ladder is used when multipath decisions are involved, where a multipath decision is a chain of ifs in the case that every statement associated with an else is an if.
+The syntax of an else if ladder:
+
+```Java
+if(condition a)
+statement-1;
+else if(condition b)
+statement-2;
+else if(condition c)
+statement-3;
+..
+..
+..
+else if(condition v)
+statement-n;
+else
+default-statement;
+statement-x;
+```
+
+The evaluation of conditions is done from the top moving to the bottom and execution happens when the true condition is found and the control transferred to statement-x.  
+The final else containing the default statement is executed when all conditions are false.
+Let us consider an example using the else if ladder:
+consider the grading of students in a school
+marks  grade
+80-100  first class
+60-79   second upper
+40-59   second lower
+0-39    pass
+the grading of students can be done using an else if ladder;
+
+```Java
+public class Main {
+  public static void main(String[] args) {
+   int marks = 22;
+if (marks > 79) {
+  System.out.println("First upper.");
+} else if (marks > 59) {
+  System.out.println("Second upper.");
+} else if(marks > 49) {
+  System.out.println("Second lower.");
+} else if(marks < 49){
+System.out.println("fail");
+}
+  }
+}
```

Comment:
The above program will work well if structured as shown below:

```
public class Main {
  public static void main(String[] args) {
   int marks = 22;

    if (marks >= 79) { //if a students marks are greater than 79
      System.out.println("First upper."); 
    } else if (marks >= 59 && marks < 79) { //if marks are greater than 59 but less than 79
      System.out.println("Second upper.");
    } else if(marks >= 49 && marks < 59) { //if marks are greater than 49 but less than 59
      System.out.println("Second lower.");
    } else if(marks < 49){
    System.out.println("fail");
    }
  }
}
```
I have made the adjustments